Improve efficiency of shape hash by not using tostring

wconstab opened the pull request:
Summary: tostring is expensive, and this is equivalent and faster

Test Plan: covered by lazy tensor unit tests

Differential Revision: D32901050

desertfire reviewed (approved):
Please land after you fix the CI errors.

alanwaketan reviewed (approved):
LGTM. May be good to indicate the name of the unit tests as well.

wconstab committed:
Improve efficiency of shape hash by not using tostring (#69496)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/69496

tostring is expensive, and this is equivalent and faster

Test Plan: covered by lazy tensor unit tests

Differential Revision: D32901050

fbshipit-source-id: 2a00d39526f2c6aa8117c3b8b9174df19d1de372
